django/db/backends/postgresql/creation.py

```diff
@@ -28,29 +28,42 @@ def sql_table_creation_suffix(self):
             template=test_settings.get('TEMPLATE'),
         )
 
+    def _execute_test_db_creation(self, cursor, parameters, keepdb=False):
+        try:
+            super()._execute_test_db_creation(cursor, parameters, keepdb)
+        except Exception as e:
+            exc_msg = 'database %s already exists' % parameters['dbname']
+            if exc_msg not in str(e):
+                # All errors except "database already exists" cancel tests
+                sys.stderr.write("Got an error creating the test database: %s\n" % e)
+                sys.exit(2)
+            elif not keepdb:
+                # if we want to keep the db, then no need to raise exception
```

Comment:
I'd change this to: `If the database should be kept, ignore "database already exists".`